hfs_fit.py

```diff
@@ -0,0 +1,717 @@
+    '''
+    if J != .5 and J != 0.:
+        return 0.5 * A * K + B * ( (3. / 4 ) * K * (K + 1) - J * (J + 1) * I * (I + 1)) / (2 * I * (2*I - 1) * J * (2*J - 1))
+    else:
+        return 0.5 * A * K
+
+#Definition and use of the saturation parameter
+def satFit(fit, sat):
+    return fit * np.exp(- sat * fit)
+
+
+class hfs:
+    def __init__(self, dataASCII = 'spectrum.txt', fitLog = 'fitLog.xlsx', nuclearSpin = 3.5):
+        '''
+        input strings can be directories to the files.
+        dataASCII is the string of directory to asc file of spectral data
+        (CSV, first column is wavenumber, second column is intensity)
+        '''
+        if fitLog != None:
+            self.fitLogName = fitLog
+            self.fitLog = pd.read_excel(self.fitLogName, index_col = 0)
+        else:
+            self.fitLogName = None
+            self.fitLog = None
+        self.I = nuclearSpin
+        print('Loading spectrum...')
+        self.dataFile = dataASCII
+        if isinstance(dataASCII, str) == True:
+            self.data = np.loadtxt(self.dataFile, delimiter = ',')
+        else:
+            print('Please use an ASCII file for the spectrum.')
+            return
+        print('Done')
+
+        self.paramsUnits = ['mK','mK','mK','mK','mK','mK','arb.','/cm','arb.']
+        self.paramsNames  = ['A_u', 'A_l', 'B_u', 'B_l', 'G_w', 'L_w', 'Area Parameter', 'CoG Wavenumber', 'Saturation Parameter']
+
+    def PlotSpec(self, wn = None):
+        '''
+        Plots the spectrum, specifying a wavenumber will plot a +- 1/cm range around it.
+        '''
+        plt.close(0)
+        plt.figure(0)
+        plt.plot(self.data[:, 0], self.data[:, 1], label = 'Spectrum')        
+        plt.xlabel(r'Wavenumber (cm$^{-1}$)')
+        plt.ylabel('Intensity (arb.)')
+        plt.title('Spectrum ' + self.dataFile)
+        if wn != None:
+            plt.xlim(wn - 1, wn + 1)
+        plt.grid()
+        plt.show()
+
+    def NewFit(self):
+        '''
+        Call this whenever you want to fit a new line.
+        Resets everything (J, A, B, Gw, etc.) so make sure you save the previous fit.
+        Have the wn range for the line and wn range for the noise ready.
+        '''
+        self.SetJNoLevList()
+        self.firstFit = True
+        self.Hold()
+        self.ParamsGuess()
+        self.lineDerivSum = self.DerivSum()
+
+    def WNRange(self, nInterp = 1):
+        '''
+        set wavenumber range for fitting
+        nInterp is number of points of even spacing between actual data to interpolate (cubic spline), set 1 for no interpolation
+        note nInterp will chagne the value from DerivSum().
+        '''
+        print('Upper Level Label: ')
+        self.upperLevel = str(input())
+        print('Lower Level Label: ')
+        self.lowerLevel = str(input())
+        self.FitDone(self.upperLevel, self.lowerLevel)
+        print('Starting wavenumber (/cm): ')
+        self.start_wn = float(input())
```

Comment:
use input makes replaying hard and testing harder, could we maybe move to a config file or somethin?